packages/create-amplify/src/package_manager.ts

```diff
@@ -0,0 +1,58 @@
+export type PackageManagerName =
+  | 'npm'
+  | 'yarn-classic'
+  | 'yarn-modern'
+  | 'pnpm';
+type PackageManagerExecutable = 'npm' | 'yarn' | 'pnpm';
+type PackageManagerBinaryRunner = 'npx' | 'yarn' | 'pnpm';
+type PackageManagerInstallCommand = 'install' | 'add';
+type PackageManagerLockFile =
+  | 'package-lock.json'
+  | 'yarn.lock'
+  | 'pnpm-lock.yaml';
+type PackageManagerInitDefault = ['init', '--yes'] | ['init'];
```

Comment:
since these unions can't be used interchangeably, I don't think there's much value here. We can just have the PackageManager objects declare these properties as strings

Reply:
We could. I just found it's easier when typing the code. If it doesn't hurt anything (thought might not be very helpful), I'd like to keep it

Reply:
Updated b491902303005be6092c2c9e66afaa7ceb0b5c5a